pandas/core/frame.py

```diff
@@ -2685,6 +2685,7 @@ def _setitem_array(self, key, value):
                 for k1, k2 in zip(key, value.columns):
                     self[k1] = value[k2]
             else:
+                self.loc._ensure_listlike_indexer(key, axis=1)
```

Comment:
it looks like you're also calling this inside _get_setitem_indexer; do you need to also call it here?

Reply:
`_get_setitem_indexer` is not called in this code path